File: examples/customer-address-jpa/src/main/java/io/zenwave360/example/core/implementation/mappers/BaseMapper.java
package io.zenwave360.example.core.implementation.mappers;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import org.mapstruct.Mapper;

@Mapper
public interface BaseMapper {

    default Instant asInstant(OffsetDateTime date) {
        return date != null ? date.toInstant() : null;
    }

    default OffsetDateTime asOffsetDateTime(Instant date) {
        return date != null ? OffsetDateTime.ofInstant(date, ZoneOffset.UTC) : null;
    }

    default LocalDateTime map(OffsetDateTime value) {
        return value != null ? value.toLocalDateTime() : null;
    }

    default OffsetDateTime map(LocalDateTime value) {
        return value != null ? OffsetDateTime.of(value, ZoneOffset.UTC) : null;
    }
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/common/mappers/BaseMapper.kt
package io.zenwave360.example.clinicaltool.common.mappers

import java.time.Instant
import java.time.LocalDateTime
import java.time.OffsetDateTime
import java.time.ZoneOffset
import org.mapstruct.Mapper

@Mapper
interface BaseMapper {

    fun asInstant(date: OffsetDateTime?): Instant? {
        return date?.toInstant()
    }

    fun asOffsetDateTime(date: Instant?): OffsetDateTime? {
        return date?.let { OffsetDateTime.ofInstant(it, ZoneOffset.UTC) }
    }

    fun map(value: OffsetDateTime?): LocalDateTime? {
        return value?.toLocalDateTime()
    }

    fun map(value: LocalDateTime?): OffsetDateTime? {
        return value?.let { OffsetDateTime.of(it, ZoneOffset.UTC) }
    }
}
